### PR TITLE
Implement aligned range actions

### PR DESCRIPTION
**Pull request by TheMaskedTurtle**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
New feature that enables to group some range actions by a groupId. Range actions sharing the same group ID would be linked. When range actions are linked a new constraint for each range action will appear and tie the set points to have the same values.
No check added on the type of range actions that are linked. So there is still the possibility to link a PST with an HVDC for example. So users have to be careful.

### CODE REVIEW

File: data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstWithRange.java [hunk 1]
```diff
@@ -48,17 +43,16 @@
...
-        super(id, name, operator, usageRules, ranges, networkElement);
+    public PstWithRange(String id, String name, String operator, List<UsageRule> usageRules, List<Range> ranges,
+                        NetworkElement networkElement, String groupId) {
+        super(id, name, operator, usageRules, ranges, networkElement, groupId);
         initAttributes();
     }
 
+    public PstWithRange(String id, String name, String operator, List<UsageRule> usageRules, List<Range> ranges, NetworkElement networkElement) {
+        super(id, name, operator, usageRules, ranges, networkElement);
+    }
```
Comment (baptsegu) [inline data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstWithRange.java:54]: I think an `initAttributes()` is needed here (all other constructors have one)

File: ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/LinearProblem.java [hunk 2]
```diff
@@ -100,6 +101,30 @@ public MPVariable getRangeActionSetPointVariable(RangeAction rangeAction) {
...
+    }
+
+    public MPConstraint addRangeActionGroupSetPointConstraint(double lb, double ub, RangeAction rangeAction) {
+        return solver.makeConstraint(lb, ub, absoluteRangeActionGroupSetPointConstraintId(rangeAction));
+    }
+
+    public MPConstraint getRangeActionGroupSetPointConstraint(RangeAction rangeAction) {
+        return solver.lookupConstraintOrNull(absoluteRangeActionGroupSetPointConstraintId(rangeAction));
+    }
+
```
Comment (baptsegu) [inline ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/LinearProblem.java:127]: I think the `absolute` in the name of the first method can be removed.

**Review by baptsegu — COMMENTED** [review]
two small comments + as said on slack, we can maybe get rid of the ElementaryRangeAction class, now that the parallel PSTs are handled with the groupId instead than being a dedicated class.

File: data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstWithRange.java (hunk 1 above)
Comment (TheMaskedTurtle) [reply data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstWithRange.java]: done

File: ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/LinearProblem.java (hunk 2 above)
Comment (TheMaskedTurtle) [reply ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/LinearProblem.java]: done

**Review by baptsegu — APPROVED** [review]
approved

**Pull request merged**